elfi/methods/bo/acquisition.py

```diff
@@ -75,61 +73,70 @@ def evaluate_gradient(self, x, t=None):
-        
-        Gaussian noise ~N(0, self.noise_cov) is added to the acquired points.
+
+        Gaussian noise ~N(0, self.noise_cov) is added to the acquired
+        points.
 
         Parameters
         ----------
-        n_values : int
-            Number of values to return.
-        pending_locations : None or numpy 2d array
-            If given, acquisition functions may
-            use the locations in choosing the next sampling
-            location. Locations should be in rows.
+        n : int
+            Number of acquisition points to return.
         t : int
-            Current iteration (starting from 0).
+            Current acq_batch_index (starting from 0).
+        random_state : np.random.RandomState, optional
 
         Returns
         -------
-        locations : 2D np.ndarray of shape (n_values, ...)
+        x : np.ndarray
+            The shape is (n_values, input_dim)
         """
-        logger.debug('Acquiring {} values'.format(n_values))
+        logger.debug('Acquiring the next batch of {} values'.format(n))
 
+        # Optimize the current minimum
         obj = lambda x: self.evaluate(x, t)
         grad_obj = lambda x: self.evaluate_gradient(x, t)
-        minloc, minval = minimize(obj, self.model.bounds, grad_obj, self.prior, self.n_inits, self.max_opt_iters)
-        x = np.tile(minloc, (n_values, 1))
-
-        # add some noise for more efficient exploration
-        if self._diagonal_cov:
-            for ii in range(self.model.input_dim):
-                if self._noise_sigma[ii] > 0:
-                    bounds_a = (self.model.bounds[ii][0] - x[:, ii]) / self._noise_sigma[ii]
-                    bounds_b = (self.model.bounds[ii][1] - x[:, ii]) / self._noise_sigma[ii]
-                    x[:, ii] = truncnorm.rvs(bounds_a, bounds_b, loc=x[:, ii], scale=self._noise_sigma[ii],
-                                             size=n_values, random_state=self.random_state)
-
-        else:
-            x += multivariate_normal.rvs(cov=self.noise_cov, size=n_values, random_state=self.random_state) \
-                 .reshape((n_values, -1))
-
-            # make sure the acquired points stay within bounds simply by clipping
-            for ii in range(self.model.input_dim):
-                x[:, ii] = np.clip(x[:, ii], *self.model.bounds[ii])
+        xhat, _ = minimize(obj, self.model.bounds, grad_obj, self.prior, self.n_inits,
+                           self.max_opt_iters, random_state=self.random_state)
+
+        # Create n copies of the minimum
+        x = np.tile(xhat, (n, 1))
+        # Add noise for more efficient fitting of GP
+        x = self._add_noise(x)
 
         return x
 
+    def _add_noise(self, x):
+        # Add noise for more efficient fitting of GP
+        if self.noise_var is not None:
+            noise_var = np.asanyarray(self.noise_var)
+            if noise_var.ndim == 0:
+                noise_var = np.tile(noise_var, self.model.input_dim)
```

Comment:
Not good to repeatedly apply the same operations on constants. I suppose there should be a setter to allow dynamic changes, if such are needed.